src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,152 @@
+            task = new Deadline(taskDetails[1], taskDetails[2]);
+            if (Integer.parseInt(taskDetails[0]) == 1) {
+                task.markAsDone();
+            }
+            break;
+        }
+        default:
+            break;
+        }
+        return task;
+    }
+
+    /**
+     * Returns an ArrayList containing task objects that were previously saved.
+     * @return ArrayList containing task objects.
+     * @throws IOException
+     */
+    public ArrayList<Task> load() throws IOException {
+        if (!Files.exists(relativeDirectoryPath)) {
+            Files.createDirectory(relativeDirectoryPath);
+        }
+        if (!Files.exists(relativeFilePath)) {
+            Files.createFile(relativeFilePath);
+        }
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner sc = new Scanner(new File(String.valueOf(relativeFilePath)));
+        ArrayList<String> stringTasks = new ArrayList<>();
+        while(sc.hasNextLine()) {
+            stringTasks.add(sc.nextLine());
+        }
+        for (String stringTask : stringTasks) {
+            String[] taskDetails = stringTask.split(" \\| ", 2);
+            Task task = null;
+            String taskType = taskDetails[0];
+            switch (taskType) {
+            case "T": {
+                task = parseStringToTask(taskDetails[1], "T");
+                break;
+            }
+            case "D": {
+                task = parseStringToTask(taskDetails[1], "D");
+                break;
+            }
+            case "E": {
+                task = parseStringToTask(taskDetails[1], "E");
+                break;
+            }
+            default:
+                break;
+            }
+            if (task != null) {
+                tasks.add(task);
+            }
+        }
+        return tasks;
+    }
+
+    /**
+     * Saves tasks to hard disk.
+     * @param tasks ArrayList containing tasks.
+     * @throws IOException
+     */
+    public void save(ArrayList<Task> tasks) {
+        try {
+            ArrayList<String> stringTasks = new ArrayList<>();
+            for (Task task : tasks) {
+                switch (task.getTaskType()) {
+                    case "T": {
+                        String data = "T" + " | " + task.getStatus() + " | " + task.getDescription();
+                        stringTasks.add(data);
+                        break;
+                    }
+                    case "D": {
+                        String data = "D" + " | " + task.getStatus() + " | " + task.getDescription();
+                        stringTasks.add(data);
+                        break;
+                    }
+                    case "E": {
+                        String data = "E" + " | " + task.getStatus() + " | " + task.getDescription();
+                        stringTasks.add(data);
+                        break;
+                    }
+                    default:
+                        break;
```

Comment:
I believe the indentation for `case` statements should be in line with the `switch` line. Not sure why the indentation is correct in `load()` but not here 🤔, perhaps consider using a code formatter and enforcing this rule?